brian2/synapses/synapses.py

```diff
@@ -794,6 +795,41 @@ def __init__(self, source, target=None, model=None, on_pre=None,
                                 'clock_driven',
                                 once=True)
                 continuous.append(single_equation)
+        def recur_check_event_summed(var,eqs):
+            """
+            Recursive Function Used to identify whether a summed variable is 
+            referring to an event-driven variable.
+
+            Parammeters
+            ----------
+            var : str
+                variable that is required for checking
+            eqs : Equations Object
+                Equations object in which we need to check
+
+            Returns
+            ----------
+            val : boolean
+                Return True if a summed variable is referring to an 
+                event-driven variable, otherwise False or continues recursion
+            """
+            val = False
+            for eq in eqs.values():
+                if var in eq.identifiers and eq.varname != var:
+                    if val == True:
+                        break
+                    elif 'summed' in eq.flags:
+                        val = True
+                        break
+                    else:
+                        val = recur_check_event_summed(eq.varname,eqs)
```

Comment:
This could be quite a bit simpler without changing the logic:
```Python
            for eq in eqs.values():
                if var in eq.identifiers and eq.varname != var:
                    if 'summed' in eq.flags:
                        return True
                    else:
                        return recur_check_event_summed(eq.varname, eqs)
            return False
```